frappe/modules/full_text_search.py

```diff
@@ -10,97 +10,219 @@
 from bs4 import BeautifulSoup
 from frappe.website.render import render_page
 from frappe.utils import set_request, cint
-from frappe.utils.global_search import get_routes_to_index
+import os
+
+class FullTextSearch:
```

Comment:
Let's break this up into separate classes. `FullTextSearch` should only deal with indexing and searching of documents.